### DISCUSSION TITLE
Does Presto read Hive cause data loss?

### DISCUSSION THREAD

**Discussion by qianmoQ** [trinodb/trino#15067] [Q&A]
Does Presto read Hive cause data loss?

When a Hive task is completed and the end time is `2022-11-16 05:43:36`, run Presto to query Hive data.
Parallel query after the end time Hive uses `select count(1)` to query the total number of data items: `342434654` Data missing `1238`
When `select count(1)` is used again 5 minutes later, the total number of query data is `342435892`

After checking, it is found that there is a block in HDFS at the time of `2022-11-16 05:43:36`, and the data inside is exactly `1238`. It is speculated that this Block is missing when Presto loads HDFS Block!

What configuration do I need to do for this situation? Or is it configured to load HDFS Block?

**Comment by willshen** [comment]
I don't think Presto/Trino read can ever cause data loss.

For your issue - do you have caching enabled that might have incomplete data at some point of time?

**Reply by qianmoQ** [reply]
The hdfs file generated after landing is the same as the query time. There is no time interval
